### ISSUE TITLE
JSONL export can publish malformed issues.jsonl after update/comment rewrites

### ISSUE THREAD

**Issue by Dicklesworthstone** [Dicklesworthstone/beads_rust#335]
## Summary

A normal Beads mutation sequence in the live `eidetic_engine_cli` tracker produced a malformed `.beads/issues.jsonl`. After the malformed export landed, global read surfaces failed with JSON parse errors, including `br show`, `br ready`, and `br dep cycles`.

This is distinct from #326. Issue #326 covers valid-but-massive JSONL churn from diagnostic/status commands. This report is about persistence integrity: mutation/export code must never replace a previously valid tracker file with invalid JSONL.

## Environment / source evidence

Observed and recorded in `/Users/jemanuel/projects/eidetic_engine_cli` local bead `bd-rzv8q`.

```text
local bead: bd-rzv8q
local title: br JSONL exporter can corrupt issues.jsonl during update/comment rewrites
first observed: 2026-06-02T22:31:38Z
current installed br in the workspace: br 0.1.13
implementation target noted by triage: /Users/jemanuel/projects/beads_rust, src/sync/mod.rs export_to_jsonl
```

The local bead records that BrownTurtle was doing tracker hygiene on 2026-06-02 and ran a normal `br update` plus `br comments add` sequence for `bd-ppbue.12`. The resulting `.beads/issues.jsonl` became malformed.

Observed corruption classes from the local evidence:

```text
- concatenated JSONL records at record boundaries, e.g. ...}{"id"... or ...}{{"id"...
- stray {"i{"id"... prefixes
- missing ]} closures for comments/dependencies arrays
- original_size:{"i... fragments where a nested issue object appeared after original_size
```

After that export, every normal tracker read failed with a parse-level configuration error, taking the whole agent tracker offline until manual repair rebuilt the file from HEAD plus the intended bead mutation.

The local bead summary describes the failure mode this way:

```text
After this happened, every br show, br ready, and br dep cycles call failed with CONFIG_ERROR: Invalid JSON at line ..., taking the whole tracker offline for agents until manual repair rebuilt the file from HEAD plus the intended bead mutation.
```

## Why this is separate from JSONL churn reports

There are now at least two adjacent writer/export hygiene failures:

- #326: diagnostic/status commands can rewrite thousands of valid JSONL records with dependency metadata churn.
- This issue: mutation/export can publish syntactically invalid JSONL, making the tracker unreadable.

The fixes may share a writer path, but parse-validity needs its own invariant. A bounded-diff check is not enough if the one changed record can still break line-level parsing; a line-count check is not enough if adjacent records collapse onto one physical line.

## Expected behavior

The JSONL export path should be atomic and parse-validated before promotion:

1. Write the candidate export to a staged file.
2. Validate every physical line parses as exactly one JSON object.
3. Validate the whole file can be imported/read by the normal command path.
4. Refuse to replace the previous valid `issues.jsonl` if any line fails.
5. Preserve the rejected candidate artifact for debugging instead of deleting it.
6. Return a structured error or degraded result naming the first bad line and the invariant that failed.

A previous valid tracker file should remain readable even if a mutation or export fails.

## Suggested regression matrix

Add fixtures or command-path tests covering mixed mutation sequences on records with comments, dependencies, labels, and optional numeric fields such as `original_size`.

The harness should reproduce or guard against these shapes:

```text
- adjacent issue records collapsed onto one line
- stray {"i prefixes before an issue object
- missing comment/dependency array closures
- original_size followed by a nested issue object
- a comment append after a status update on the same issue
- a dependency rewrite followed by comment append/export
```

Recommended invariants:

```text
- exactly one JSON object per physical line
- each line parses independently
- no issue object appears inside another issue field
- comments[] and dependencies[] arrays are closed before the record boundary
- full-file import/read succeeds after update + comment + dependency operations
- previous valid issues.jsonl is preserved when candidate export validation fails
```

## Acceptance criteria

- A focused regression fails against the old writer/export behavior represented by `bd-rzv8q`.
- `br update` + `br comments add` + dependency/export scenarios cannot produce invalid JSONL.
- Export validates the candidate file before replacing `issues.jsonl`.
- On validation failure, `br` preserves both the previous valid JSONL and the rejected candidate, and reports a structured first-failure diagnosis.
- `br show`, `br ready`, and `br dep cycles --json` continue to work after the mutation matrix completes.
- Recovery guidance documents how to inspect/preserve malformed candidate artifacts without deleting `.beads` files.

## Related issues

- #326: valid-but-massive diagnostic/status JSONL churn.
- #328: merge-artifact warning needs machine-actionable classification.
- #334: umbrella authority/claimability contract for safe agents.

**Comment by Dicklesworthstone** [comment]
Fixed. Export now writes a temp file then `durable_rename`s it into place (`sync/mod.rs:2256-2258`), gated by a pre-promotion verification: `count_issues_in_jsonl(temp)` runs `analyze_jsonl`, which `serde_json::from_str`-parses **every** line and errors ("Invalid JSON at line N") on malformed content, plus a count-match assertion (`sync/mod.rs:2231-2258`). A `}{`-style concatenation fails the parse before the rename, so the live `issues.jsonl` is never replaced with invalid JSONL; `TempFileGuard` cleans up on failure. This is full per-line parse validation, not just a line-count check.
